subprojects/language-native/src/integTest/groovy/org/gradle/language/cpp/CppIncrementalCompileIntegrationTest.groovy

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.cpp
+
+import groovy.io.FileType
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.app.IncrementalCppStaleCompileOutputApp
+import org.gradle.nativeplatform.fixtures.app.IncrementalCppStaleCompileOutputLib
+
+class CppIncrementalCompileIntegrationTest extends AbstractInstalledToolChainIntegrationSpec {
```

Comment:
Should add some test coverage for when all of the source files are removed.

Reply:
Thanks for point this out. I'm doing this "all source are removed" scenario as part of another PR that test specifically stale link output.